etna/transforms/encoders/mean_segment_encoder.py

```diff
@@ -46,9 +50,27 @@ def transform(self, df: pd.DataFrame) -> pd.DataFrame:
         -------
         :
             result dataframe
+
+        Raises
+        ------
+        ValueError:
+            If transform isn't fitted.
+        ValueError:
+            If there are segments that weren't present during training.
         """
+        if self.global_means is None:
+            raise ValueError("The transform isn't fitted!")
+
+        segments = df.columns.get_level_values("segment").unique().tolist()
+        new_segments = set(segments) - self.global_means.keys()
+        if len(new_segments) > 0:
+            raise ValueError(
+                f"This transform can't process segments that weren't present on train data: {new_segments}"
```

Comment:
Maybe we shouldn't include `new_segments` in error message. If one passes a completely different dataset with a large number of segments, the message will be cluttered.

Reply:
May be we should try: https://docs.python.org/3/library/reprlib.html
But I think that in most places we are already printing all broken segments.

Reply:
Ok, let's leave as is right now and consider this case in future.

Reply:
I fixed this already.